src/LiveComponent/src/LiveDownloadResponse.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Symfony\UX\LiveComponent;
+
+use SplFileInfo;
+use SplTempFileObject;
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\HeaderUtils;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class LiveDownloadResponse extends BinaryFileResponse
+{
+    public const HEADER_LIVE_DOWNLOAD = 'X-Live-Download';
+
+    public function __construct(string|SplFileInfo $file, ?string $filename = null)
+    {
+        if (\is_string($file)) {
+            $file = new SplFileInfo($file);
+        }
+
+        if ((!$file instanceof SplFileInfo)) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not exist.', $file));
+        }
```

Comment:
I'm not sure to understand when this code can be executed, `$file` can either be a `string` or `SplFileInfo`, and when it's a `string`, we re-assign `$file` to a `SplFileInfo`.

Reply:
Maybe the condition is wrong?